### PR TITLE
Fix station filter logic for nodes having no children.

### PR DESCRIPTION
**Pull request by junghao**
As requested in https://github.com/GeoNet/tickets/issues/12671.
Also, text output will be in sync with XML output https://github.com/GeoNet/help/issues/103.

Previously we deliberately make it not exists, so the fix is remove those extras.

### CODE REVIEW

File: cmd/fdsn-ws/fdsn_station.go [hunk 1]
```diff
@@ -618,10 +626,11 @@ func (r *FDSNStationXML) doFilter(params []fdsnStationV1Search) bool {
...
 
+	nNet := len(r.Network)
 	r.Network = ns
 
-	if len(ns) == 0 {
-		// No result ( no "Network" node )
+	if nNet != 0 && len(ns) == 0 {
+		// this means the query has network parameter but idn't match one, we want to give a no data
 		return false
 	}
```
Comment (CallumNZ) [inline cmd/fdsn-ws/fdsn_station.go:635]: `nNet` is strange.
Can `ns` be `networks` ? or better yet `filteredNetworks`.

Would a `defer r.Network = ns` help, just after for loop? Also for station/channel sections.

Comment missing a `d` in `didn't`

**Review by CallumNZ — CHANGES_REQUESTED** [review]
Just some variable names that are hard to understand.

File: cmd/fdsn-ws/fdsn_station.go (hunk 1 above)
Comment (junghao) [reply cmd/fdsn-ws/fdsn_station.go]: There's still `trimLevel` need to operate on the result slice. So can't move the assignment to defer.

**Review by CallumNZ — DISMISSED** [review]

**Comment by junghao** [comment]
@CallumNZ  I found that the logic was incorrect. Could you please review my new update?

Thanks

**Review by CallumNZ — APPROVED** [review]
Looks good.

File: cmd/fdsn-ws/fdsn_station.go [hunk 2]
```diff
@@ -663,14 +665,20 @@ func (n *NetworkType) doFilter(params []fdsnStationV1Search) bool {
...
 	if len(n.Station) == 0 {
 		// for network nodes without children (though unlikely to happen):
-		//   1. If the query parameter stops at network level, then the match is done
-		//   2. Otherwise, we're unable to get a match because of empty children (thus returning false)
+		//   If there are query parameters for furthur levels, and there's no "*" (match anything) in the parameters,
+		//     then it'll be impossible to find a match (because we don't have children)
 		for _, p := range matchedParams {
-			if p.StationReg == nil && p.ChannelReg == nil && p.LocationReg == nil {
-				return true
+			if p.StationReg != nil && !contains(p.StationReg, REGEX_ANYTHING) {
```
Comment (CallumNZ) [inline cmd/fdsn-ws/fdsn_station.go:671]: What does the regex check add?

Comment (junghao) [reply cmd/fdsn-ws/fdsn_station.go:671]: See Jerome's comment regarding issue with WTSZ here https://github.com/GeoNet/tickets/issues/12671#issuecomment-1452655370

**Review by CallumNZ — APPROVED** [review]
Looks good

**Pull request merged**